ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/PeerSelection/MockEnvironment.hs

```diff
@@ -796,14 +869,70 @@ instance Arbitrary GovernorMockEnvironment where
+             (\(ArbitraryLedgerStateJudgement lsj) -> do
+                  (praosKnown, genesisKnown) <- (,) <$> knownGen <*> knownGen
+                  (praosRootKnown, genesisRootKnown) <- (,) <$> rootKnownGen praosKnown <*> rootKnownGen genesisKnown
+                  (praosEst, genesisEst) <- (,) <$> estGen praosKnown <*> estGen genesisKnown
+                  (praosAct, genesisAct) <- (,) <$> actGen praosEst <*> actGen genesisEst
+                  (praosBigKnown, Positive genesisBigKnown)
+                    <- (,) <$> resize 1000 arbitrarySizedNatural
+                           <*> resize 1000 arbitrary `suchThat` ((>= 10) . getPositive)
+                  (praosBigEst, genesisBigEst) <- (,) <$> choose (0, min 1000 praosBigKnown)
+                                                      <*> choose (1, min 1000 genesisBigKnown)
+                  (praosBigAct, genesisBigAct) <- (,) <$> choose (0, min 100 praosBigEst)
+                                                      <*> choose (1, min 100 genesisBigEst)
+
+                  let targets0 =
+                        ConsensusModePeerTargets {
+                          praosTargets = PeerSelectionTargets {
+                              targetNumberOfRootPeers = praosRootKnown,
+                              targetNumberOfKnownPeers = praosKnown,
+                              targetNumberOfEstablishedPeers = praosEst,
+                              targetNumberOfActivePeers = praosAct,
+                              targetNumberOfKnownBigLedgerPeers = praosBigKnown,
+                              targetNumberOfEstablishedBigLedgerPeers = praosBigEst,
+                              targetNumberOfActiveBigLedgerPeers = praosBigAct },
+                          genesisSyncTargets = PeerSelectionTargets {
+                              targetNumberOfRootPeers = genesisRootKnown,
+                              targetNumberOfKnownPeers = genesisKnown,
+                              targetNumberOfEstablishedPeers = genesisEst,
+                              targetNumberOfActivePeers = genesisAct,
+                              targetNumberOfKnownBigLedgerPeers = genesisBigKnown,
+                              targetNumberOfEstablishedBigLedgerPeers = genesisBigKnown,
+                              targetNumberOfActiveBigLedgerPeers = genesisBigAct } }
+                  let lsjWithDelay = (,) lsj <$> elements [ShortDelay, NoDelay]
+                      -- synchronize target basis with ledger state judgement
+                      -- so we can use tests which check if the right targets
+                      -- are selected
+                      targetsWithDelay = (,) targets0 <$> case consensusMode of
+                                                     PraosMode -> elements [ShortDelay, NoDelay]
+                                                     GenesisMode -> snd <$> lsjWithDelay
+                  (,) <$> lsjWithDelay <*> targetsWithDelay)
+           <&> bimap wrap wrap . unzip
```

Comment:
This is quite hard to read, I would perhaps move the locally defined functions to a `where` clause to de-clutter some space

Reply:
Done